core/cmd/client.go

```diff
@@ -270,6 +270,23 @@ func (n ChainlinkRunner) Run(ctx context.Context, app chainlink.Application) err
 	return errors.WithStack(g.Wait())
 }
 
+func logErrorAndRepeat(ctx context.Context, lggr logger.Logger, cfg config.GeneralConfig, fn func() error) {
+	for {
+		// try calling fn() and log error if any
+		if err := fn(); err != nil {
+			lggr.Criticalf("%v", err)
```

Comment:
What about the expected ErrServerClosed on a clean shutdown?

Reply:
Fixed